stack.yaml

```diff
@@ -0,0 +1,66 @@
+# This file was automatically generated by 'stack init'
```

Comment:
[Stack](https://www.stackage.org/) is a project that's tried to solve the "cabal hell" issue by curating known-working package lists. In practice, this yaml file just serves to tell the stack executable what versions of the compiler and whatnot to use.

I take it as a big confidence boost in the "compilability" of a project I am looking at using, but I can remove this all from the PR if you'd like